zcash_client_backend/src/data_api/ll.rs

```diff
@@ -0,0 +1,554 @@
+    /// note or UTXO.
+    ///
+    /// If the wallet database has stored a wallet address that contains the given receiver, then
+    /// that address is returned; otherwise, a new address containing that receiver is generated.
+    fn select_receiving_address(
+        &self,
+        account: Self::AccountId,
+        receiver: &Receiver,
+    ) -> Result<Option<ZcashAddress>, Self::Error>;
+
+    /// Detects and returns the identifier for the account that generated the given address, if the
+    /// address belongs to an account in the wallet.
+    ///
+    /// In addition, for HD-derived addresses, the change-level key scope used to derive the
+    /// address is returned, so that the caller is able to determine whether any special handling
+    /// rules apply to the address for the purposes of preserving user privacy (by limiting address
+    /// linking, etc.).
+    #[cfg(feature = "transparent-inputs")]
+    #[allow(clippy::type_complexity)]
+    fn find_account_for_transparent_address(
+        &self,
+        address: &TransparentAddress,
+    ) -> Result<Option<(Self::AccountId, Option<TransparentKeyScope>)>, Self::Error>;
+
+    /// Detects the set of accounts that received transparent outputs corresponding to the provided
+    /// [`OutPoint`]s. This is used to determine which account(s) funded a given transaction.
+    ///
+    /// [`OutPoint`]: transparent::bundle::OutPoint
+    #[cfg(feature = "transparent-inputs")]
+    fn detect_accounts_transparent<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a transparent::bundle::OutPoint>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Detects the set of accounts that received Sapling outputs that, when spent, reveal(ed) the
+    /// given [`Nullifier`]s. This is used to determine which account(s) funded a given
+    /// transaction.
+    ///
+    /// [`Nullifier`]: sapling::Nullifier
+    fn detect_accounts_sapling<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a sapling::Nullifier>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Detects the set of accounts that received Orchard outputs that, when spent, reveal(ed) the
+    /// given [`Nullifier`]s. This is used to determine which account(s) funded a given
+    /// transaction.
+    ///
+    /// [`Nullifier`]: orchard::note::Nullifier
+    #[cfg(feature = "orchard")]
+    fn detect_accounts_orchard<'a>(
+        &self,
+        spends: impl Iterator<Item = &'a orchard::note::Nullifier>,
+    ) -> Result<HashSet<Self::AccountId>, Self::Error>;
+
+    /// Get information about a transparent output controlled by the wallet.
+    ///
+    /// # Parameters
+    /// - `outpoint`: The identifier for the output to be retrieved.
+    /// - `spendable_as_of`: The target height of a transaction under construction that will spend the
+    ///   returned output. If this is `None`, no spendability checks are performed.
```

Comment:
This is a confusing parameter name to me. The backend wallet will have its own view of the chain tip, and thus of whether the outpoint is spendable or not; the target of the `spendable` adjective in the name is unclear. If you don't like `target_height` for this, then maybe `check_if_spendable_as_of` to clarify it?